htdocs/projet/class/task.class.php

```diff
@@ -1329,6 +1329,79 @@ public function getListContactId($source = 'internal')
 		return $contactAlreadySelected;
 	}
 
+	/**
+	 * Merge contact of tasks
+	 *
+	 * @param 	int 	$origin_id 	Old task id
+	 * @param 	int 	$dest_id 	New task id
+	 * @return 	bool
+	 */
+	public function mergeContactTask($origin_id, $dest_id)
+	{
+		$error = 0;
+		$origintask = new Task($this->db);
+		$result = $origintask->fetch($origin_id);
+		if ($result <= 0) {
+			return false;
+		}
+
+		//Get list of origin contacts
+		$arraycontactorigin = array_merge($origintask->liste_contact(-1, 'internal'), $origintask->liste_contact(-1, 'external'));
+		foreach ($arraycontactorigin as $key => $contact) {
+			$result = $this->add_contact($contact["id"], $contact["fk_c_type_contact"], $contact["source"]);
+			if ($result < 0) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Merge time spent of tasks
+	 *
+	 * @param 	int 	$origin_id 	Old task id
+	 * @param 	int 	$dest_id 	New task id
+	 * @return 	bool
+	 */
+	public function mergeTimeSpentTask($origin_id, $dest_id)
+	{
+		global $user;
+
+		$error = 0;
+		$origintask = new Task($this->db);
+		$result = $origintask->fetch($origin_id);
+		if ($result <= 0) {
+			return false;
+		}
+
+		$result = $origintask->fetchTimeSpentOnTask();
+		if ($result <= 0) {
+			return false;
+		}
+		$timespentorigin = $origintask->lines;
+		foreach ($timespentorigin as $key => $timespent) {
+			$origintask->timespent_id = $timespent->timespent_line_id;
+
+			$this->timespent_date = $timespent->timespent_line_date;
+			$this->timespent_datehour = $timespent->timespent_line_datehour;
+			$this->timespent_withhour = $timespent->timespent_line_withhour;
+			$this->timespent_duration = $timespent->timespent_line_duration;
+			$this->timespent_fk_user = $timespent->timespent_line_fk_user;
+			$this->timespent_fk_product = $timespent->fk_project;
+			$this->timespent_note = $timespent->timespent_line_note;
+
+			$result = $this->addTimeSpent($user);
```

Comment:
Adding and deleting has a lot of bad effect. Risk to forgot a field if a new field is added in future, mode code. More insert, update.
Why not just updating the task id from old value to new one with a simple update ?

Reply:
@eldy  would the same apply to merging contacts and members?

Reply:
> @eldy would the same apply to merging contacts and members?

Currently, we can merge only thirdparties. But being able to merge a contact with another contact or a memeber with another one, would be very interesting... Code must follow same method than merging thirdparties.